File: frontend/lib/models/course_model.dart
class CourseModel {
  final int? id;

  final String name;

  final String description;

  final String classCode;
  final int? ownerId;
  final int maxMembers;

  CourseModel({
    this.id,
    required this.name,
    required this.description,
    required this.classCode,
    required this.maxMembers,
    this.ownerId,
  });

  factory CourseModel.fromJson(
      Map<String, dynamic> json) {

    return CourseModel(

      id: json["id"],

      name: json["name"] ?? "",

      description:
      json["description"] ?? "",

      classCode:
      json["classCode"] ?? "",
      maxMembers: json["maxMembers"] ?? 30,
      ownerId: json["ownerId"],
    );
  }

  Map<String, dynamic> toJson() {

    return {

      "name": name,

      "description":
      description,
      "maxMembers": maxMembers,
    };
  }
}

File: backend/src/models/course.model.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const Classroom = sequelize.define(
  "Classroom",
  {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true,
    },

    name: {
      type: DataTypes.STRING(255),
      allowNull: false,
    },

    description: {
      type: DataTypes.TEXT,
      allowNull: true,
    },

    classCode: {
      type: DataTypes.STRING(10),
      allowNull: false,
      unique: true,
    },

    ownerId: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },

    // Thêm mới
    maxMembers: {
      type: DataTypes.INTEGER,
      allowNull: false,
      defaultValue: 30,
    },
  },
  {
    tableName: "Classrooms",
    timestamps: true,
  }
);

module.exports = Classroom;